metabase/metabase
# Support basic theming in interactive data tables for embedding SDK

Closes https://github.com/metabase/metabase/issues/42158

### Description

We want the data table to inherit the user's theming options (e.g. `brand` color), and provide a great out-of-the-box experience by default. We also want to give user more flexibility in customizing the design of their data tables, in case they need more control (e.g. non-white background).

![CleanShot 2567-05-11 at 00 21 26@2x](https://github.com/metabase/metabase/assets/4714175/ae505ff4-2e13-4f62-9a7d-547233d643a1)

### How to verify

- Navigate to the `kitchen-sink-data-table` branch on the demo app
- Go to the kitchen sink.
- Go to `AppProvider.tsx` and change the colors in the `colors` and `components` section:
  - Comment out the entire `components` block. It should follow the `brand` color.
  - Uncomment the `components` block. It should follow `table.cell` and `table.idColumn`
     - Cell hover color should be a lighter color of cell background if it's a dark color, and a transparent brand color if cell background is a light color.
- Start the Metabase local instance
   - Data table should remain the same in the webapp

### Caveat

- The header color cannot be customized in this PR. It will always be white, for now. We plan to add this option in later PRs.

### Demo

Without any theme configuration, it follows the `brand` color.

![CleanShot 2567-05-13 at 16 42 23@2x](https://github.com/metabase/metabase/assets/4714175/54c6bac6-5a74-4bc1-a692-b039fb0aa452)

With `cell.textColor`, `cell.backgroundColor` and `idColumn.backgroundColor` set

![CleanShot 2567-05-13 at 16 44 40@2x](https://github.com/metabase/metabase/assets/4714175/b98824fd-e6dd-4592-951f-9c84ca466c0d)

No changes to the existing interactive data table in the dashboard.

![CleanShot 2567-05-13 at 16 47 07@2x](https://github.com/metabase/metabase/assets/4714175/4d5e252e-45c1-4313-9acc-0fff3bee4f95)

### Checklist

   - I might add more tests to verify that themes from the SDK are being applied to the table's elements, but let's do this in a separate PR as this one is getting a bit large.
